managing-os/templates/fedora/upgrade-27-to-28.md

```diff
@@ -0,0 +1,423 @@
+      file /usr/lib/python2.7/site-packages/xcffib/xinput.pyo from install of python2-xcffib-0.5.1-5.fc28.noarch conflicts with file from package python-xcffib-0.5.1-1.fc26.noarch
+
+To work around this error:
+
+1. Upgrade everything besides `python2-xcffib` and `qubes-core-vm` (use `-x python2-xcffib -x qubes-core-vm`).
+2. Upgrade `python2-xcffib` using `sudo dnf swap python-xcffib python2-xcffib`. This should automatically update `qubes-gui-vm` too.
+
+
+Qubes 3.2 Instructions
+----------------------
+
+### Summary: Upgrading the Standard Fedora 27 Template to Fedora 28 ###
+
+**Note:** The prompt on each line indicates where each command should be entered
+(`@dom0` or `@fedora-28`).
+
+        [user@dom0 ~]$ qvm-clone fedora-27 fedora-28
+        [user@dom0 ~]$ truncate -s 5GB /var/tmp/template-upgrade-cache.img
+        [user@dom0 ~]$ qvm-run -a fedora-28 gnome-terminal
+        [user@dom0 ~]$ qvm-block -A fedora-28 dom0:/var/tmp/template-upgrade-cache.img
+        [user@fedora-28 ~]$ sudo mkfs.ext4 /dev/xvdi
+        [user@fedora-28 ~]$ sudo mount /dev/xvdi /mnt/removable
+        [user@fedora-28 ~]$ sudo dnf clean all
+        [user@fedora-28 ~]$ sudo dnf --releasever=28 --setopt=cachedir=/mnt/removable --best --allowerasing distro-sync
+
+    (Shut down TemplateVM by any normal means.)
+
+        [user@dom0 ~]$ rm /var/tmp/template-upgrade-cache.img
+        [user@dom0 ~]$ qvm-trim-template fedora-28
+
+(Optional cleanup: Switch everything over to the new template and delete the old
+one. See instructions below for details.)
+
+
+### Detailed: Upgrading the Standard Fedora 27 Template to Fedora 28 ###
+
+These instructions will show you how to upgrade the standard Fedora 27
+TemplateVM to Fedora 28. The same general procedure may be used to upgrade any
+template based on the standard Fedora 27 template.
+
+**Note:** The command-line prompt on each line indicates where each command
+should be entered (`@dom0` or `@fedora-28`).
+
+ 1. Ensure the existing template is not running.
+
+        [user@dom0 ~]$ qvm-shutdown fedora-27
+
+ 2. Clone the existing template and start a terminal in the new template.
+
+        [user@dom0 ~]$ qvm-clone fedora-27 fedora-28
+        [user@dom0 ~]$ qvm-run -a fedora-28 gnome-terminal
+
+ 3. Attempt the upgrade process in the new template.
+
+        [user@fedora-28 ~]$ sudo dnf clean all
+        [user@fedora-28 ~]$ sudo dnf --releasever=28 distro-sync --best --allowerasing
+
+    **Note:** `dnf` might ask you to approve importing a new package signing
+    key. For example, you might see a prompt like this one:
+
+        warning: /var/cache/dnf/fedora-d02ca361e1b58501/packages/python2-babel-2.3.4-1.fc28.noarch.rpm: Header V3 RSA/SHA286 Signature, key ID f5282ee4: NOKEY
+        Importing GPG key 0xF5282EE4:
+         Userid     : "Fedora (28) <fedora-28-primary@fedoraproject.org>"
+         Fingerprint: 860E 19B0 AFA8 00A1 7518 81A6 F55E 7430 F528 2EE4
```

Comment:
Fingerprint for Fedora 28 is `128C F232 A937 1991 C8A6  5695 E08E 7E62 9DB6 2FB1`